spanner/test/cloudexecutor/executor/internal/inputstream/handler.go

```diff
@@ -84,16 +104,57 @@ func (h *CloudStreamHandler) Execute() error {
 			log.Printf("Failed to receive request from client: %v", err)
 			return err
 		}
+		actionType := getActionType(req.Action)
+		if actionType == "Read" || actionType == "Query" {
+			readOrQueryActionPresent = true
+		}
+
 		if err = h.startHandlingRequest(ctx, req); err != nil {
 			log.Printf("Failed to handle request %v, Client ends the stream with error: %v", req, err)
 			// TODO(sriharshach): should we throw the error here instead of nil?
 			return nil
 		}
 	}
+	// End the top-level OpenTelemetry span.
+	trace.EndSpan(ctx, nil)
+
+	// OpenTelemetry trace created for a streaming request will be verified using Cloud Trace APIs to make sure that
+	// Spanner server side tracing is working. Check will be performed only if there is atleast one action of type
+	// "Read" or "Query" in streaming request, trace is sampled and number of checks performed are less than the max limit.
+	if h.CloudTraceCheckAllowed && ottrace.SpanContextFromContext(ctx).IsSampled() && readOrQueryActionPresent {
+		h.serverSideTraceCheckDone = true
+		// Create a trace client to read the traces.
+		traceClient, err := traceapiv1.NewClient(ctx, h.TraceClientOptions...)
+		if err != nil {
+			return fmt.Errorf("Failed to create trace client: %v", err)
+		}
+		defer func() {
+			// Close the trace client.
+			log.Println("Closing the trace client")
+			traceClient.Close()
+		}()
+
+		// Verify the end to end trace exported to Cloud Trace.
+		traceId := ottrace.SpanContextFromContext(ctx).TraceID().String()
+		if err = verifyCloudTraceExportedTraces(ctx, traceClient, traceId); err != nil {
+			log.Printf("Verification failed for exported traces: %v", err)
+			return err
+		}
+	}
 	log.Println("Done executing actions")
```

Comment:
Lets move this line after `trace.EndSpan(ctx, nil) `,
This way we can know that action completed successfully and any error further might be due to e2e tracing

Reply:
Okay. makes sense. Done